src/profile-proxy-actions/profile-proxy-actions.db.ts

```diff
@@ -0,0 +1,71 @@
+import { PROFILE_PROXY_ACTIONS_TABLE } from '../constants';
+import { ProfileProxyActionEntity, ProfileProxyActionType } from '../entities/IProfileProxyAction';
+import {
+  ConnectionWrapper,
+  dbSupplier,
+  LazyDbAccessCompatibleService
+} from '../sql-executor';
+
+export class ProfileProxyActionsDb extends LazyDbAccessCompatibleService {
+  async insertProfileProxyAction({
+    profileProxyAction,
+    connection
+  }: {
+    readonly profileProxyAction: ProfileProxyActionEntity;
```

Comment:
I would change it to `Omit< ProfileProxyActionEntity, 'id' | 'created_at' ... and all other fields that are not decidable by creator>` and create those two fields inside this function, not in the service. It just doesn't make sense to let external classes decide those fields.